src/data/fin/basic.lean

```diff
@@ -570,6 +572,10 @@ def cast_lt (i : fin m) (h : i.1 < n) : fin n := ⟨i.1, h⟩
 
 @[simp] lemma cast_lt_mk (i n m : ℕ) (hn : i < n) (hm : i < m) : cast_lt ⟨i, hn⟩ hm = ⟨i, hm⟩ := rfl
 
+lemma cast_lt_cast_lt {m n : ℕ} (i : fin n) (hm : i.val < m) (hn : i.val < n) :
+  (i.cast_lt hm).cast_lt hn = i :=
+by simp [fin.cast_lt]
```

Comment:
Can you prove the more general `(i.cast_lt hm).cast_lt hn = i.cast_lt hn` instead?

Reply:
Do you mean `(i.cast_lt hm).cast_lt hn = i.cast_lt hk`, with three different numbers?